universal-application-tool-0.0.1/app/services/question/QuestionDefinition.java

```diff
@@ -109,6 +108,19 @@ public QuestionType getQuestionType() {
     return ImmutableMap.of("text", ScalarType.STRING);
   }
 
+  /** Get a map of scalars stored by this question definition. */
+  @JsonIgnore
+  public ImmutableMap<String, ScalarType> getScalars(boolean includeFullPath) {
+    ImmutableMap<String, ScalarType> scalars = this.getScalars();
+    if (includeFullPath) {
+      ImmutableMap.Builder<String, ScalarType> ret = new ImmutableMap.Builder<String, ScalarType>();
```

Comment:
not sure if this exists but maybe you could use a bistream. this is pseudocode...

`BiStream.from(scalars).mapKeys(k -> this.getPath() + "." + k).toMap();`

Reply:
it do exist https://google.github.io/mug/apidocs/com/google/mu/util/stream/BiStream.html